annotation-utils/src/main/java/eu/europeana/annotation/utils/parse/AnnotationLdParser.java

```diff
@@ -731,6 +764,275 @@ private void setLatitude(Body body, String latitude) throws JsonParseException {
+		if (body instanceof AgentBody) {
+			if (valueObject instanceof JSONObject) {	
+				Map<String, String> resPlaceOfDeath = extractStringStringMapFromJsonObject(valueObject);	
+				((AgentBody) body).setPlaceOfDeath(resPlaceOfDeath);
+			} else {
+				Map<String, String> resPlaceOfDeath = extractStringStringMapFromSerializedStringMap(valueObject);
+				((AgentBody) body).setPlaceOfDeath(resPlaceOfDeath);
+			}
+		} else {
+			throw new JsonParseException("placeOfDeath not supported for bodyType: " + body.getInternalType());
+		}
+	}
+
+    /**
+     * @param body
+     * @param valueObject
+     * @throws JsonParseException
+     * @throws JSONException
+     */
+    private void setDateOfBirth(Body body, Object valueObject) throws JsonParseException, JSONException {		
+		if (body instanceof AgentBody) {
+			if (valueObject instanceof JSONArray) {	
+				List<String> resList = extractStringListFromJsonArray(valueObject);
+				((AgentBody) body).setDateOfBirth(resList);
+			} else {
+				List<String> resList = extractStringListFromSerializedStringArray(valueObject);
+				((AgentBody) body).setDateOfBirth(resList);
+			}
+		} else {
+			throw new JsonParseException("dateOfBirth not supported for bodyType: " + body.getInternalType());
+		}
+	}
+
+	/**
+	 * This method extracts a list of strings from serialized JSON array. 
+	 * In this use case we have only one array element.
+	 * @param valueObject
+	 * @return list of strings
+	 * @throws JSONException
+	 */
+	private List<String> extractStringListFromJsonArray(Object valueObject) throws JSONException {
+		JSONArray arr = (JSONArray) valueObject;
+		List<String> resList = new ArrayList<String>();
+		for(int i = 0; i < arr.length(); i++){
+		    resList.add(arr.getString(i));
+		}
+		return resList;
+	}
+
+    /**
+     * @param body
+     * @param valueObject
+     * @throws JsonParseException
+     * @throws JSONException
+     */
+    private void setDateOfDeath(Body body, Object valueObject) throws JsonParseException, JSONException {		
+		if (body instanceof AgentBody) {
+			if (valueObject instanceof JSONArray) {	
+				List<String> resList = extractStringListFromJsonArray(valueObject);
+				((AgentBody) body).setDateOfDeath(resList);
+			} else {
+				List<String> resList = extractStringListFromSerializedStringArray(valueObject);
+				((AgentBody) body).setDateOfDeath(resList);
+			}
+		} else {
+			throw new JsonParseException("dateOfDeath not supported for bodyType: " + body.getInternalType());
+		}
+	}
+
+	/**
+	 * This method extracts a list of strings from serialized string array. 
+	 * In this use case we have only one array element.
+	 * @param valueObject The JSON object
+	 * @return list of strings
+	 */
+	private List<String> extractStringListFromSerializedStringArray(Object valueObject) {
+		List<String> resList = new ArrayList<String>();
+		String strValueObject = valueObject.toString();
+		String value = strValueObject.substring(1, strValueObject.length()-1);
```

Comment:
It is already parsed. The code just removed braces. But it works also with braces. Custom parsing removed now.

Reply:
Removed the method and serialization. If we have only one element in the list e.g. for dateOfBirth, then JsonLd automatically does string instead of array.